src/components/layouts/php-collection-page-layout.tsx

```diff
@@ -0,0 +1,1189 @@
+            </div>
+            <div className="text-xs text-gray-700 dark:text-gray-500">
+              {convertTimeWithTitles(lesson.duration, {
+                showSeconds: true,
+              })}
+            </div>
+          </div>
+        )}
+      </div>
+    </li>
+  )
+}
+
+const ModuleCollection = ({module, children}: {module: any; children: any}) => {
+  return (
+    <div key={module.id}>
+      <h2 className="text-xl font-bold mt-4 mb-2">{module.title}</h2>
+      {(() => {
+        switch (true) {
+          case module.contentList.length < 5:
+            return (
+              <div className={'grid grid-cols-2'}>
+                <ul className="grid gap-4">{children}</ul>
+                <p className="mb-6 prose text-gray-900 dark:prose-dark md:prose-lg md:dark:prose-lg-dark dark:text-gray-100 dark:prose-a:text-blue-300 dark:hover:prose-a:text-blue-200 prose-a:text-blue-500 hover:prose-a-:text-blue-600 ">
+                  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed
+                  do eiusmod tempor incididunt ut labore et dolore magna aliqua.
+                  Ut enim ad minim veniam, quis nostrud exercitation ullamco
+                  laboris nisi ut aliquip ex ea commodo consequat. Duis aute
+                  irure dolor in reprehenderit in voluptate velit esse cillum
+                  dolore eu fugiat nulla pariatur. Excepteur sint occaecat
+                  cupidatat non proident, sunt in culpa qui officia deserunt
+                  mollit anim id est laborum.
+                </p>
+              </div>
+            )
+          case module.contentList.length > 5 && module.contentList.length <= 10:
+            return (
+              <div>
+                <p className="mb-6 prose text-gray-900 dark:prose-dark md:prose-lg md:dark:prose-lg-dark dark:text-gray-100 dark:prose-a:text-blue-300 dark:hover:prose-a:text-blue-200 prose-a:text-blue-500 hover:prose-a-:text-blue-600 ">
+                  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed
+                  do eiusmod tempor incididunt ut labore et dolore magna aliqua.
+                  Ut enim ad minim veniam, quis nostrud exercitation ullamco
+                  laboris nisi ut aliquip ex ea commodo consequat. Duis aute
+                  irure dolor in reprehenderit in voluptate velit esse cillum
+                  dolore eu fugiat nulla pariatur. Excepteur sint occaecat
+                  cupidatat non proident, sunt in culpa qui officia deserunt
+                  mollit anim id est laborum.
+                </p>
+                <ul className="grid grid-rows-5 grid-flow-col gap-4">
+                  {children}
+                </ul>
+              </div>
+            )
+          default:
+            return <ul className="grid grid-cols-1">{children}</ul>
+        }
+      })()}
+    </div>
+  )
+}
+
+const CollectionContent = (props: any) => {
+  return props.contentList.map((content: any, index: number) => {
```

Comment:
Incoming collections from sanity should have a "content list" array where other collections and resources may live

Reply:
The idea here is that we can support whatever type of content we have within a collection using recursion and switching on the content type. Child "collections" will recurse and then the types of their content will be handled

Reply:
traditionally "items" has been the name I've given or seen used for the actual list. 

my problem with contents list is that the items might not be "content"

another option might be "resources" which would be aligned with our collection/resources language

https://www.dropbox.com/s/qg4ppej4az9sulg/egghead%20Templates%20v4.pdf?dl=0

reading this and the description of the "standard list" in the context of a collection, i'd probably lean "resources"

I don't feel super strongly about this and consistency is probably the most important going forawrd!